src/services/Discounts.php

```diff
@@ -541,12 +541,14 @@ public function matchOrder(Order $order, Discount $discount): bool
             return false;
         }
 
-        $orderDiscountConditionParams = [
-            'order' => $order->toArray([], ['lineItems.snapshot', 'shippingAddress', 'billingAddress'])
-        ];
+        if ($discount->orderConditionFormula) {
```

Comment:
@nfourtythree this was a massive performance improvement. Serialising the order every time we matched a discount was very expensive.